core/domain/user_domain.py

```diff
@@ -345,3 +345,79 @@ def validate(self):
             raise utils.ValidationError(
                 'Expected can_review_questions to be a boolean value, '
                 'found: %s' % type(self.can_review_questions))
+
+
+class ModifiableUserData(python_utils.OBJECT):
+    """Domain object to represent change in allowed attributes for UserSettings
+    and UserAuthDetails domain object.
+    """
+
+    def __init__(
+            self, display_alias, pin, preferred_language_codes,
+            preferred_site_language_code, preferred_audio_language_code,
+            version, user_id=None):
+        """Constructs a ModifiableUserData domain object.
+
+        Args:
+            display_alias: str. Display alias of the user shown on Android.
+            pin: str or None. PIN of the user used for PIN based authentication
+                on Android. None if it hasn't been set till now.
+            preferred_language_codes: list(str) or None. Exploration language
+                preferences specified by the user.
+            preferred_site_language_code: str or None. System language
+                preference.
+            preferred_audio_language_code: str or None. Audio language
+                preference.
+            version: int. The version of modifiable user data schema.
+            user_id: str or None. User ID of the user whose data is being
+                updated. None if request did not have a user_id for the user
+                yet and expects the backend to create a new user entry for it.
+        """
+        self.display_alias = display_alias
+        self.pin = pin
+        self.preferred_language_codes = preferred_language_codes
+        self.preferred_site_language_code = preferred_site_language_code
+        self.preferred_audio_language_code = preferred_audio_language_code
+        # The user_id is not intended to be a modifiable attribute, it is just
+        # needed to identify the object.
+        self.user_id = user_id
+        self.version = version
```

Comment:
@seanlip A minor follow up:
In the merged PR I decided to **keep** the `version` attribute. Schema version is another class attribute and remains constant, but the version is passed during initialisation. Version is needed in case schema version updates to > 1 and we try to convert from older version to latest version.

My previous comment and the finally merged code (that keeps version) contradict each other and since the comment wasn't resolved I thought of bringing it to your notice.

Reply:
Thanks for bringing this up, I might have missed this.  We should follow the existing paradigms, e.g. the Exploration class in exp_domain.py. Note that that is a versioned model, so a "version" is part of the domain object initialization. However, a "schema version" is not.

So, two things:

(a) Make sure to carefully distinguish between version and schema version.
(b) For this case, if we only have a schema version, then why do you actually need the "version" field? What does it represent? (In the Exploration domain object, the version and schema version are different things.)

Reply:
(Note that the payload from Android would still have a schema version, which you can then use in order to identify the correct conversion function to use. But probably you'll want that schema version field to be "outside" the rest of the processed dict.)

Reply:
@seanlip That for clearing it out. I won't be needing `version` attribute, the schema version is enough. I will correct this in the issue #10736 itself (because that's where I realised this mistake as it would be used in controllers).

Reply:
Sounds good. Thanks!